generate.xml

```diff
@@ -2060,6 +2079,7 @@
         <option value="--disable-static" />
         <option value="--build-boost" />
         <option value="--build-zmq" />
+        <option value="--build-mbedtls" />
```

Comment:
For this work in server, this is a new requirement that's always on (i.e. there isn't a configuration that works without it).  Either that needs to change, or we need a travis build setup that has a system install of mbedtls.  Preference?

Reply:
This should be made an optional dependency/feature if possibke, but even without that we should verify the valid scenario of using the existing dependency.

Reply:
Hrm, the optional approach is doable, but not as cleanly as we'd like (lots of `#ifdef` nonsense most likely).  By config, it can be enabled/disabled as needed, but as a build dependency, not at the moment.  I'm not sure how to make travis have mbedtls in the base system, but otherwise agree that it would be good to test that case.

Reply:
All of the optional builds require conditional compilation. The trick is to minimize their impact on source readability.